manifests/config/server/globalnamingresources.pp

```diff
@@ -0,0 +1,126 @@
+# Definition: tomcat::config::server::globalnamingresources
+#
+# Configure Resource elements in $CATALINA_BASE/conf/server.xml
+#
+# Parameters:
+# - $catalina_base is the base directory for the Tomcat installation.
+# - $connector_ensure specifies whether you are trying to add or remove the
+#   Connector element. Valid values are 'true', 'false', 'present', and
+#   'absent'. Defaults to 'present'.
+# - The $port attribute. This attribute is required unless $connector_ensure
+#   is set to false.
+# - The $protocol attribute. Defaults to $name when not specified.
+# - $parent_service is the Service element this Connector should be nested
+#   beneath. Defaults to 'Catalina'.
+# - An optional hash of $additional_attributes to add to the Connector. Should
+#   be of the format 'attribute' => 'value'.
+# - An optional array of $attributes_to_remove from the Connector.
+define tomcat::config::server::globalnamingresources (
+  $resource_name,
+  $auth,
+  $type,
+  $factory,
+  $description     = undef,
+  $driverClassName = undef,
+  $maxActive       = undef,
+  $maxIdle         = undef,
+  $maxWait         = undef,
+  $username        = undef,
+  $url             = undef,
+  $password        = undef,
+  $pathname        = undef,
+  $catalina_base   = $::tomcat::catalina_home,
+  $resource_ensure = 'present',
+) {
+  if versioncmp($::augeasversion, '1.0.0') < 0 {
+    fail('Server configurations require Augeas >= 1.0.0')
+  }
+
+  validate_re($resource_ensure, '^(present|absent|true|false)$')
+
+  $base_path = "Server/GlobalNamingResources/Resource[#attribute/name='${resource_name}']"
+
+  if $resource_ensure =~ /^(absent|false)$/ {
+    $changes = "rm ${base_path}"
+  } else {
+    $_resource_name   = "set ${base_path}/#attribute/name ${resource_name}"
+    $_auth            = "set ${base_path}/#attribute/auth ${auth}"
+    $_type            = "set ${base_path}/#attribute/type ${type}"
+    $_factory         = "set ${base_path}/#attribute/factory ${factory}"
+
+    if $description {
+      $_description = "set ${base_path}/#attribute/description '${description}'"
+    }
+    else {
+      $_description = undef
+    }
```

Comment:
for one, this is not the way we write if/else… at least not according to the [style guide](https://docs.puppetlabs.com/guides/style_guide.html), it is according to the [language doc](http://docs.puppetlabs.com/puppet/3.7/reference/lang_conditional.html#if-statements) but anyway, all of those should *still* be [selectors](http://docs.puppetlabs.com/puppet/3.7/reference/lang_conditional.html#selectors)